lib/auth/webauthn/login.go

```diff
@@ -370,7 +373,7 @@ func (f *loginFlow) finish(ctx context.Context, user string, resp *wantypes.Cred
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	if credential.Authenticator.CloneWarning {
+	if credential.Authenticator.CloneWarning && !challengeAllowReuse {
```

Comment:
The counter "repeats" because the MFA response is reused, but after first use storage is updated and expects to see an incremented counter? Is that correct?

I don't think we want to fully ignore the warning, but maybe we could accept an off-by-one SignCount error if reuse is at play.

Let's also test this scenario before we apply any fixes.

Reply:
Maybe something like this?

```go
	if credential.Authenticator.CloneWarning {
		// Reused challenges trigger clone warnings because, after first use, we expect
		// the counter to be up by one.
		isReuseCounterMismatch := challengeAllowReuse &&
			len(u.credentials) == 1 /* sanity check */ &&
			credential.Authenticator.SignCount != u.credentials[0].Authenticator.SignCount-1
		if !isReuseCounterMismatch {
			log.WarnContext(ctx, "Clone warning detected for device, the device counter may be malfunctioning",
				"user", user,
				"device", dev.GetName(),
			)
		}
	}
```

Reply:
That works as well, tested it.